torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -765,7 +765,41 @@ void addInputs(
 }
 
 void addInputs(Node* n, const char* name, ArrayRef<double> value) {
-  AT_ERROR("Tracing float lists currently not supported!");
+  AT_ERROR("XXX");
+  // using ArgumentStash = jit::tracer::ArgumentStash;
+  // std::vector<Value*> info = ArgumentStash::hasDoubleArrayRef(name)
+  //     ? ArgumentStash::popDoubleArrayRef(name)
+  //     : ArgumentStash::DoubleArrayRefTrace(value.size());
+
+  // auto& g = getTracingState()->graph;
+  // for (size_t i = 0; i < info.size(); ++i) {
+  //   if (info[i] != nullptr)
+  //     continue;
+  //   info[i] = g->insertConstant(value[i]);
+  //   recordSourceLocation(info[i]->node());
+  // }
+  // for (jit::Value* v : info) {
+  //   if (*v->type() != *jit::FloatType::get()) {
+  //     throw std::runtime_error(
+  //         "Type mismatch in setposattr for ArrayRef<double>. Check that your program "
+  //         "is valid without tracing, and please file a bug report if it is.");
+  //   }
+  // }
+  // n->addInput(
+  //     g->insertNode(g->createList(jit::FloatType::get(), info))->output());
```

Comment:
Can anyone give me some guidance on how to implement this?  Is ArgumentStash relevant here?  Do I need to mirror all of those `IntArrayRef` methods for doubles?

Reply:
I don't think ArgumentStash is relevant. iirc the place where it's relevant is when we switch return values for things like `sizes()` to be Tensors during tracing so we can record them. Downstream, someone might use those Tensors in a list that specifies sizes or something, and because the int values contained in those Tensors are the results of `sizes()`, we want to record them because they can affect the flow of the program.

OTOH, I don't think any float-valued stuff participates in that special-case recording behavior, so I think you can omit that stuff here.